substrate/client/transaction-pool/tests/integration.rs

```diff
@@ -0,0 +1,55 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Testsuite of fatp integration tests.
+
+pub mod zombienet;
+
+use std::time::Duration;
+
+use tokio::join;
+use zombienet::NetworkSpawner;
+use zombienet_sdk::subxt::OnlineClient;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn send_future_and_then_ready_from_single_account() {
+	let _ = env_logger::try_init_from_env(
+		env_logger::Env::default().filter_or(env_logger::DEFAULT_FILTER_ENV, "info"),
+	);
+	let net = NetworkSpawner::init_from_asset_hub_fatp_low_pool_limit_spec().await.unwrap();
+	let collator = net.get_node("charlie").unwrap();
+	let _client: OnlineClient<zombienet_sdk::subxt::SubstrateConfig> =
+		collator.wait_client_with_timeout(120u64).await.unwrap();
+	let ws = "ws://127.0.0.1:9933";
+	let mut nonce = 0;
+	for _ in 0..3 {
+		// Spawn future TXs.
+		let future_start = nonce + 5;
+		let handle1 = tokio::spawn(async move {
+			cmd_lib::run_cmd!(RUST_LOG=info ttxt tx --chain=sub --ws=$ws from-single-account --account 0 --count 5 --from $future_start)
+		});
+		tokio::time::sleep(Duration::from_secs(5)).await;
```

Comment:
I know it is kinda temporary - ideally if we could avoid timeouts. They cause a lot of troubles in CI.

I see two options:
- we could ask nodes if they have enough txs in the pool - unluckily we don't have API for this. See discussion around: https://github.com/paritytech/polkadot-sdk/issues/7138
- but we could also add some status getter to ttxt API and check if txs were sent (or validated). (could be sth like `get_validate_count`). This seems to be a mid-term way to go.

Reply:
I agree we should avoid the sleep because of the CI + the non-deterministic way the spawned tasks might be polled behind the scenes by the tokio runtime.

What you suggested sounds useful, but maybe we can implement them for more advanced scenarios. I would avoid complicating this initial scenario just because I think we should be fine assessing future and ready txs sent in parallel, in a high number. You suggested offline that we can use the `from-many-accounts`, and just to clarify more concretely, I would use an account range for ttxt between `0..99`, with future txs starting from nonce `100`, with a count of `100`, so 10k future txs, and ready txs for the same accounts range, and with nonce starting at `0`, with count `100`.

Reply:
Implemented the above scenario and it works based on this PR: https://github.com/michalkucharczyk/tx-test-tool/pull/22. #7257 is pending now on the  `tx-test-tool` PR, to get merged, `ttxt` to be published as a lib on crates.io, and then the path dependency for ttxt to be replaced with a crates.io one.

Reply:
Resolving this, we implemented the test here and also polished ttxt.